File: apps/web/lib/api/dplayer-comment.ts
/**
 * DPlayer 互換コメント形式
 * @see https://dplayer.js.org/
 */
export interface DPlayerComment {
  time: number;

  /**
   * コメント種類
   * - "normal": 中央スクロール
   * - "top"/"ue": 上部表示
   * - "bottom"/"shita": 下部表示
   */
  type: 'normal' | 'top' | 'bottom';

  size: 'small' | 'normal' | 'big';

  /**
   * コメント色（16進数カラーコード）
   */
  color: string;

  author?: string | null;

  text: string;
}

/**
 * DPlayer 互換コメント一覧レスポンス
 */
export interface DPlayerCommentListResponse {
  comments: DPlayerComment[];
}

File: apps/desktop/lib/models/dplayer_comment.dart
class DPlayerComment {
  final num time;
  final String type;
  final String size;
  final String color;
  final String? author;
  final String text;

  DPlayerComment({
    required this.time,
    required this.type,
    required this.size,
    required this.color,
    this.author,
    required this.text,
  });

  factory DPlayerComment.fromJson(Map<String, dynamic> json) {
    return DPlayerComment(
      time: json['time'] as num,
      type: json['type'] as String? ?? 'normal',
      size: json['size'] as String? ?? 'normal',
      color: json['color'] as String? ?? '#ffffff',
      author: json['author'] as String?,
      text: json['text'] as String,
    );
  }
}
